src/cargo/core/resolver/errors.rs

```diff
@@ -214,9 +215,13 @@ pub(super) fn activation_error(
     let all_req = semver::VersionReq::parse("*").unwrap();
     let mut new_dep = dep.clone();
     new_dep.set_version_req(all_req);
-    let mut candidates = match registry.query_vec(&new_dep, false) {
-        Ok(candidates) => candidates,
-        Err(e) => return to_resolve_err(e),
+
+    let mut candidates = loop {
+        match registry.query_vec(&new_dep, false) {
+            Ok(Poll::Ready(candidates)) => break candidates,
+            Ok(Poll::Pending) => (), // TODO: dont hot loop for it to be Ready
```

Comment:
Since this is just an error reporting context, could we presumably ignore pending here?

Reply:
Found a way to make that happen.